src/main/java/org/wso2/carbon/esb/connector/auth/AmazonLambdaAuthConnector.java

```diff
@@ -0,0 +1,438 @@
+                    /**
+                     * Checks for the parameter which contains integer value. If it is such parameters then
+                     * it directly put the value without putting inside quotation mark while building the payload.
+                     */
+                    else if (key.equals(AmazonLambdaConstants.API_PUBLISH) ||
+                            key.equals(AmazonLambdaConstants.API_TIMEOUT)) {
+                        payloadBuilder.append(payloadValues)
+                                .append(',');
+                    }
+                    /**
+                     *If both conditions mentioned above fails then the values should be inside the quotation mark.
+                     * This condition does so.
+                     */
+                    else {
+                        payloadBuilder.append('"');
+                        payloadBuilder.append(payloadValues);
+                        payloadBuilder.append('"');
+
+                        payloadBuilder.append(',');
+                    }
+                }
+            }
+            /**
+             * Checks the length of payload if it is greater than zero, meaning payload is not empty, it keeps all the
+             * appended payload's parameters and values within the "{}".
+             */
+            if (payloadBuilder.length() > 0) {
+                requestPayload = "{" + payloadBuilder.substring(0, payloadBuilder.length() - 1) + "}";
+            }
+            /**
+             * Setting requestPayload to the message context which will be used by methods to send the payload while
+             * making the API request.
+             */
+            messageContext.setProperty(AmazonLambdaConstants.REQUEST_PAYLOAD, requestPayload);
+            //Hashing and making it lowercase hexadecimal string for appending to canonical request.
+            canonicalRequest.append(bytesToHex(hash(messageContext, requestPayload)).toLowerCase());
+            //Creates stringToSign
+            stringToSign.append(AmazonLambdaConstants.AWS4_HMAC_SHA_256);
+            stringToSign.append(AmazonLambdaConstants.NEW_LINE);
+            stringToSign.append(amzDate);
+            stringToSign.append(AmazonLambdaConstants.NEW_LINE);
+            stringToSign.append(dateOnly);
+            stringToSign.append(AmazonLambdaConstants.FORWARD_SLASH);
+            stringToSign.append(messageContext.getProperty(AmazonLambdaConstants.REGION));
+            stringToSign.append(AmazonLambdaConstants.FORWARD_SLASH);
+            stringToSign.append(messageContext.getProperty(AmazonLambdaConstants.SERVICE));
+            stringToSign.append(AmazonLambdaConstants.FORWARD_SLASH);
+            stringToSign.append(messageContext.getProperty(AmazonLambdaConstants.TERMINATION_STRING));
+            stringToSign.append(AmazonLambdaConstants.NEW_LINE);
+            stringToSign.append(bytesToHex(hash(messageContext, canonicalRequest.toString())).toLowerCase());
```

Comment:
You can append values to stringToSign like
stringToSign.append().append()

Reply:
Fix with 408c7bd